src/netbsd.rs

```diff
@@ -1,15 +1,30 @@
 //! Implementation for NetBSD
-use crate::{lazy::LazyPtr, util_libc::sys_fill_exact, Error};
-use core::{ffi::c_void, mem::MaybeUninit, ptr};
+use crate::{util_libc::sys_fill_exact, Error};
+use core::{
+    ffi::c_void,
+    mem::MaybeUninit,
+    ptr,
+    sync::atomic::{AtomicPtr, Ordering},
+};
+
+unsafe extern "C" fn polyfill_using_kern_arand(
+    buf: *mut u8,
+    buflen: libc::size_t,
+    flags: libc::c_uint,
+) -> libc::ssize_t {
+    debug_assert_eq!(flags, 0);
 
-fn kern_arnd(buf: &mut [MaybeUninit<u8>]) -> libc::ssize_t {
     static MIB: [libc::c_int; 2] = [libc::CTL_KERN, libc::KERN_ARND];
-    let mut len = buf.len();
+
+    // NetBSD will only return up to 256 bytes at a time, and
+    // older NetBSD kernels will fail on longer buffers.
+    let mut len = cmp::max(buflen, 256);
```

Comment:
Sorry, I meant `min(buflen, 256)` (I edited the above comment after posting).

Reply:
Hi, thanks. I wasn't expecting people to look at this already but I did address this.